operator/operator.go

```diff
@@ -438,7 +459,19 @@ func (o *Operator) rescaleStatefulSet(ctx context.Context, sts *appsv1.StatefulS
 	}
 
 	if len(pods.Items) > replicas {
+		log.Infof("Starting pod draining from %d to %d pods", len(pods.Items), replicas)
 		for _, pod := range pods.Items[replicas:] {
+			// first, check if we need to opt-out of the loop because the EDS changed.
+			newSR, err := srg.Get(ctx)
+			if err != nil {
+				return fmt.Errorf("failed to refresh EDS: %v", err)
+			}
+			newDesiredReplicas := int(newSR.Replicas())
+			if newDesiredReplicas > desiredReplicas {
+				log.Infof("EDS %s/%s target scaling definition changed from %d to %d, aborting scale-down", sr.Namespace(), sr.Name(), desiredReplicas, newDesiredReplicas)
```

Comment:
probably more clear if we use `newSR` here instead of `sr`. Same when calling `Drain` on line 488 below.

Reply:
fixed